tsl/test/shared/sql/constify_timestamptz_op_interval.sql

```diff
@@ -0,0 +1,11 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+
+-- all of these should have all chunk exclusion happening at plan time
+EXPLAIN SELECT time FROM metrics WHERE time < '2000-01-01'::timestamptz - '6h'::interval;
+EXPLAIN SELECT time FROM metrics WHERE time < '2000-01-01'::timestamptz - '1 day'::interval;
+EXPLAIN SELECT time FROM metrics WHERE time < '2000-01-07'::timestamptz - '7 day'::interval;
+-- month calculation are not constified so exclusion happens at execution time
+EXPLAIN SELECT time FROM metrics WHERE time < '2000-02-01'::timestamptz - '1 month'::interval;
+
```

Comment:
I'm missing a lot of strange combinations. With the exception of the last case, these are all positives. Can you add tests for the following:

* INTERVAL + TIMESTAMP
* TIMESTAMP - INTERVAL + INTERVAL
* TIMESTAMP + INTERVAL '1 month' - INTERVAL '1 day'
* TIMESTAMP + INTERVAL '1 month' + INTERVAL '1 day'
* INTERVAL '1 month' + TIMESTAMP - INTERVAL '1 day'
* INTERVAL '1 month' + TIMESTAMP + INTERVAL '1 day'
* TIMESTAMP - INTERVAL '1 month - 3 days'
* TIMESTAMP - INTERVAL '1 month - 3 days'

Reply:
i'm gonna add more tests of course but none of the tests you are suggested are targeted by this PR so i'm not gonna add any of these suggestions

Reply:
Well...

```
mats=# explain verbose select interval '3 days' + timestamptz '2020-06-01 10:11:12';
                                     QUERY PLAN                                      
-------------------------------------------------------------------------------------
 Result  (cost=0.00..0.01 rows=1 width=8)
   Output: ('2020-06-01 10:11:12+02'::timestamp with time zone + '3 days'::interval)
(2 rows)
```

Reply:
I would assume that your PR will capture the rewritten expression?

Reply:
TIMESTAMP != TIMESTAMPTZ

Reply:
Ah, when I wrote TIMESTAMP below, I meant in the generic sense. Please replace with `TIMESTAMPTZ`. :)